### PR TITLE
Fix worker spec deviations

### PR DESCRIPTION
**Pull request by rdhananjaya**
## Purpose

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/29843
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/30149
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/30146
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/18024

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java [hunk 1]
```diff
@@ -3863,20 +3924,168 @@ private void validateWorkerInteractions(WorkerActionSystem workerActionSystem) {
...
+    }
+
+    private List<String> getWorkerNameList(BLangExpression expr, SymbolEnv functionEnv) {
+        ArrayList<String> workerNames = new ArrayList<>();
+        populateWorkerNameList(expr, workerNames, functionEnv);
+        return workerNames;
+    }
+
+    private void populateWorkerNameList(BLangExpression expr, ArrayList<String> workerNames, SymbolEnv functionEnv) {
+        if (expr.getKind() == NodeKind.BINARY_EXPR) {
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java:4055]: what is the case where we get a Binary expression?

Comment (rdhananjaya) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java:4055]: It's alternate wait action, I think I have missed to add negative tests for them, will add (`wait w|v`)

**Review by hasithaa — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java (hunk 1 above)
Comment (hasithaa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java:4055]: yes, let's add that as a separate PR. I will merge this for now. Please create an issue for that.

Comment (rdhananjaya) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java:4055]: Created https://github.com/ballerina-platform/ballerina-lang/issues/30285

**Pull request merged**